src/lib/core/error/error-options.ts

```diff
@@ -7,27 +7,24 @@
  */
 
 import {InjectionToken} from '@angular/core';
-import {FormControl, FormGroupDirective, NgForm} from '@angular/forms';
+import {FormGroupDirective, NgForm, NgControl} from '@angular/forms';
 
 /** Injection token that can be used to specify the global error options. */
 export const MD_ERROR_GLOBAL_OPTIONS = new InjectionToken<ErrorOptions>('md-error-global-options');
 
 export type ErrorStateMatcher =
```

Comment:
SGTM, but because the matcher is really just the function, it would look more along the lines of this:

```
export class ErrorOptions {
  isErrorState = (control, form) => control.invalid && (form.submitted || control.dirty);
  // more error-related config goes here in the future.
}
```

Also I'm not sure whether we should provide the `ShowOnDirtyErrorStateMatcher` by default since it's pretty basic logic that people can write themselves if they're overriding it anyway.